src/Calypso-SystemPlugins-SUnit-Browser/ClyJumpToTestMethodCommand.class.st

```diff
@@ -0,0 +1,98 @@
+"
+Description
+--------------------
+
+I am a command available in the method context menu of Calypso. When activated I'll jump on the test method corresponding to the method. 
+
+If the method does not exists I will create it in the corresponding class.
+
+Internal Representation and Key Implementation Points.
+--------------------
+
+    Instance Variables
+	selectedMethods:			<aCollection>	Methods selected by the user that should have their tests generated.
+	testMethodToBrowse:		<aMethod>			The method to browse at the end of the execution of the command.
+
+"
+Class {
+	#name : #ClyJumpToTestMethodCommand,
+	#superclass : #CmdCommand,
+	#traits : 'TClyGenerateTestClass',
+	#classTraits : 'TClyGenerateTestClass classTrait',
+	#instVars : [
+		'selectedMethods',
+		'testMethodToBrowse'
+	],
+	#category : #'Calypso-SystemPlugins-SUnit-Browser-Commands'
+}
+
+{ #category : #testing }
+ClyJumpToTestMethodCommand class >> canBeExecutedInContext: aBrowserContext [
+	^ aBrowserContext isInstanceSideMethodSelected and: [ aBrowserContext selectedMethods anySatisfy: [ :each | each isTestMethod not ] ]
```

Comment:
Also direct messages to objects are unfriendly for remote scenario. It will produce remote message send for each method when menu will be opened.
Replace it with code from ClyRunTestsFromSelectedItemsCommand class>>#canBeExecutedInContext:

Reply:
So we need a new method property and we need to add it in ClySUnitEnvironmentPlugin>>#decorateBrowserItem:ofClass:? Or is it in ClySUnitEnvironmentPlugin>>#collectMetadataOfMethods:?

Reply:
Existing properties are enough. You just need to use them instead of #isTestMethod:
```Smalltalk
canBeExecutedInContext: aBrowserContext [
	^ aBrowserContext isInstanceSideMethodSelected and: [ 
              aBrowserContext selectedItems anySatisfy: [ :each | 
			(each hasProperty: ClyTestResultProperty) not ]]
```

Reply:
Ok, great. I'll do that soon (tonight or tomorrow).

The name "TestResult" confused me a little since we have "TestResult" in SUnit. I was not expecting this name to tag test methods.

Reply:
In remote scenario it allows to send test result together with test methods.